### PR TITLE
Cura 4 update

### PR DESCRIPTION
**Pull request by karl-nilsson**
Preliminary work before the official 4.0 release

### CODE REVIEW

File: srcpkgs/Uranium/template [hunk 1]
```diff
@@ -14,5 +15,6 @@ short_desc="Python framework for building Desktop applications"
 maintainer="Karl Nilsson <karl.robert.nilsson@gmail.com>"
 license="LGPL-3.0-or-later"
 homepage="https://github.com/Ultimaker/Uranium"
-distfiles="https://github.com/Ultimaker/Uranium/archive/${version}.tar.gz"
-checksum=9a63e7d02c57a818815f61ada53b41d6c04683f874db80baa32f68d34b51a245
+distfiles="https://github.com/Ultimaker/Uranium/archive/${version}-beta.tar.gz"
+checksum=755cc8def46e1f4884d23bb85762144eac0c057333702e517986df644d6da864
+
```
Comment (Cogitri) [inline srcpkgs/Uranium/template]: That newline doesn't belong there, does it?

File: srcpkgs/libArcus/template [hunk 2]
```diff
@@ -1,18 +1,19 @@
 # Template file for 'libArcus'
 pkgname=libArcus
-version=3.6.0
-revision=2
+version=4.0
+revision=1
 build_style=cmake
 configure_args="-DBUILD_EXAMPLES=OFF"
+wrksrc="${pkgname}-${version}-beta"
```
Comment (Cogitri) [inline srcpkgs/libArcus/template]: wrksrc should be before build_style

File: srcpkgs/Uranium/template [hunk 3]
```diff
@@ -1,9 +1,10 @@
 # Template file for 'Uranium'
 pkgname=Uranium
-version=3.6.0
-revision=2
+version=4.0
+revision=1
 archs=noarch
 build_style=cmake
+wrksrc="${pkgname}-${version}-beta"
```
Comment (Cogitri) [inline srcpkgs/Uranium/template]: Same here

File: srcpkgs/libSavitar/template [hunk 4]
```diff
@@ -1,16 +1,17 @@
 # Template file for 'libSavitar'
 pkgname=libSavitar
-version=3.6.0
+version=4.0
 revision=1
 build_style=cmake
+wrksrc="${pkgname}-${version}-beta"
 hostmakedepends="python3"
```
Comment (Cogitri) [inline srcpkgs/libSavitar/template]: Same here then

**Comment by karl-nilsson** [comment]
@Cogitri fixed
the wkrsrc additions were to deal with the "-beta" name
I should be able to remove them when the 4.0 release becomes official

**Comment by Cogitri** [comment]
Yup, thought it was worth mentioning nontheless :)

**Comment by karl-nilsson** [comment]
Cura v4.0.0 released, testers welcome!

**Comment by Gigahawk** [comment]
Can't seem to build cura-engine, xbps-src never finds libArcus-devel even after I've manually built it. It always just tries to build it again, and then errors out with `Unable to locate 'libArcus-devel-4.0.0_1' in repository pool.`

**Comment by Gigahawk** [comment]
Looks like there's an issue with libArcus, cura crashes when trying to import it.

http://ix.io/1E0N

**Comment by Gigahawk** [comment]
`cura` needs `-DCURA_SDK_VERSION=6.0.0` added to `configure_args` as per this [issue](https://github.com/Ultimaker/Cura/issues/5142).

`Uranium` needs the following patch to make calling it from the command line work as per this [issue](https://github.com/Ultimaker/Cura/issues/5290)
```
From 237dc959ca536ae4a45d1ba59d4ad6cf86a36946 Mon Sep 17 00:00:00 2001
From: Ghostkeeper <rubend@tutanota.com>
Date: Mon, 25 Feb 2019 11:58:02 +0100
Subject: [PATCH] Fix install prefix when installing in symlinked directory

If /bin is a symlink to /usr/bin and /bin appears before /usr/bin on the user's PATH, it would get sys.argv[0]='/bin/python3'. This then messes up the install location ('/') which consequently messes up the data and config storage directories.
This solution is provided by CapnKernel and should resolve the symlink so that sys.argv[0]='/usr/bin/python3' again and the storage data and config storage directories become something like /usr/share/... again.
---
 UM/Application.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git UM/Application.py UM/Application.py
index 05adb8f3..f709438a 100644
--- UM/Application.py
+++ UM/Application.py
@@ -398,9 +398,10 @@ class Application:
     @staticmethod
     def getInstallPrefix() -> str:
         if "python" in os.path.basename(sys.executable):
-            return os.path.abspath(os.path.join(os.path.dirname(sys.argv[0]), ".."))
+            executable = sys.argv[0]
         else:
-            return os.path.abspath(os.path.join(os.path.dirname(sys.executable), ".."))
+            executable = sys.executable
+        return os.path.abspath(os.path.join(os.path.dirname(os.path.realpath(executable)), ".."))
 
     __instance = None   # type: Application
 
-- 
2.21.0

```

**Comment by Gigahawk** [comment]
Also looks like the plugin marketplace doesn't work. Maybe we need to set the CURA_CLOUD_API cmake variables?

**Comment by karl-nilsson** [comment]
@Gigahawk thanks, I'll work on it tonight

**Comment by karl-nilsson** [comment]
@Gigahawk I fixed the issues you mentioned, with the exception of the plugin marketplace. Do you have any output or screencaps you can share that demonstrate the breakage?

**Comment by Gigahawk** [comment]
[Initial screen when clicking marketplace button](https://imgur.com/aWUNZtH)

[Looks like the plugin manager still works for plugins I've already installed from older versions of cura](https://imgur.com/OvQH2o2)

After I click the installed tab I can't go back to either the plugins or the materials tab.
They look like they're styled differently so I guess they're disabled because Cura can't connect to whatever it's supposed to connect to.

**Comment by Gigahawk** [comment]
Looks like setting `-DCURA_CLOUD_API_ROOT=https://api.ultimaker.com` and `-DCURA_CLOUD_API_VERSION=1` [fixes the marketplace issue](https://imgur.com/a/JMRNBIf).

**Comment by karl-nilsson** [comment]
@Gigahawk 
It should be fixed, please try again.

Even if the CLOUD_API variables aren't set in cmake, they are set to sane defaults [here](https://github.com/Ultimaker/Cura/blob/4.0/cura/UltimakerCloudAuthentication.py), with the exception of CURA_CLOUD_API_VERSION, [which has been fixed in master.](https://github.com/Ultimaker/Cura/commit/72a7cc59cee47301ddbe5cbc861fcad6bd26e589#diff-32dcefc2274011b1548e60531103c857)

I opted to simply add CURA_CLOUD_API_VERSION to configure_args, rather than include the 2-line patch, but the other CURA_CLOUD variables shouldn't need this.

Thanks!

**Comment by Gigahawk** [comment]
Yup, seems to work fine. This is probably ready to merge.

**Pull request merged**